src/lib.rs

```diff
@@ -40,38 +45,66 @@ pub extern fn whatlang_detect(ptr: *const c_char, cinfo: &mut CInfo) -> u8 {
     }
 }
 
+fn detect_internal(text: &str, cinfo: &mut CInfo) -> u8 {
```

Comment:
Since both users of `detect_internal` actually have &[u8] available, and do the same handling for utf8 parsing, maybe this function should take &[u8], and handle parsing into a str and returning `2` itself?